pkg/controller/sync/dispatch/managed.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dispatch
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
+)
+
+// DispatchResource is the subset of the FederatedResource interface
+// required for dispatching operations to managed resources.
+type DispatchResource interface {
```

Comment:
[Suggestion] What say about naming this `ResourceToDispatch`. I had to wrap my head around some portion of the rest of the code to understand that `DispatchResource` is not an action (literally speaking).

Reply:
Also if this is a subset of FederatedResource, then why not embed this in FederatedResource rather then duplicating the method definitions.

Reply:
Done.